### PR TITLE
[Parent][MBL-14320] Calendar Api Migration

### PR DESCRIPTION
**Pull request by Blackchai**
Okay, first, couple of behavior changes:

-Announcements, ungraded discussions, and ungraded quizzes do not show up in the calendar.
-Calendar filter behavior changes:
   -Calendar filter has max of 10 items (shows error)
   -Calendar filter has min of 1 item (shows error)
   -Calendar filter defaults to selecting first 10

That should cover the changes. Other than that, the calendar should function exactly the same. Items routing, items displaying submission status, etc. Also, there shouldn't be any need for "migration". If the user was previously selecting 0 courses (the old "all"), this will default them to the first 10. If they were previously selecting more than 10, it will be trimmed down to 10.

One last note, I've attempted to leave in sections of code that might make it easy for us to migrate back if the planner api is updated. I'm worried that reverting this commit might be too challenging if sufficient time has passed. I'm willing to remove those comments though if folks disagree.

Jk, one more "last" note. I'm not particularly confident of how I wrote the "serialize" portion of the custom SubmissionWrapper. Deserialization is definitely working, for both the grades list case (a list of submissions) and the calendar/student case (single submission). But we aren't currently serializing it anywhere, so I just have my tests...

### CODE REVIEW

File: apps/flutter_parent/lib/screens/calendar/calendar_widget/calendar_filter_screen/calendar_filter_list_screen.dart [hunk 1]
```diff
@@ -104,7 +108,10 @@ class CalendarFilterListScreenState extends State<CalendarFilterListScreen> {
                   )
                 : _courseList(_courses);
           } else {
-            selectedContextIds.addAll(widget._selectedCourses);
+            // This is a user with a filter from before the api migration, make sure we trim their list down.
+            selectedContextIds.addAll(widget._selectedCourses.length > 10
```
Comment (JordanMarshall) [inline apps/flutter_parent/lib/screens/calendar/calendar_widget/calendar_filter_screen/calendar_filter_list_screen.dart:112]: Would `widget._selectedCourses.take(10)` work here?

File: apps/flutter_parent/lib/models/submission_wrapper.dart [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+class SubmissionWrapperSerializer implements StructuredSerializer<SubmissionWrapper> {
+  @override
+  SubmissionWrapper deserialize(Serializers serializers, Iterable serialized,
+      {FullType specifiedType = FullType.unspecified}) {
+    // Initialize an empty builder
+    final result = new SubmissionWrapperBuilder();
+
+    if (serialized.first == 'id') {
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/flutter_parent/lib/models/submission_wrapper.dart:52]: Ohhh good point.

File: apps/flutter_parent/lib/models/schedule_item.dart [hunk 3]
```diff
@@ -92,7 +95,87 @@ abstract class ScheduleItem implements Built<ScheduleItem, ScheduleItemBuilder>
...
+  String _parseContextType(String code) {
+    final index = code.indexOf('_');
+    return code.substring(0, index);
+  }
+
+  PlannerSubmission getPlannerSubmission() {
+    if (assignment == null) return null;
+
+    final submission = assignment?.submissionList?.submission;
+    if (submission == null) return PlannerSubmission((b) => b..submitted = false);
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/flutter_parent/lib/models/schedule_item.dart:152]: So I believe the only case where we get the submissionList is assignment submissions for assignment groups (as those aren't specific to one student, but the observee)... But I'll do some testing. Wouldn't hurt to check it as well.

File: apps/flutter_parent/lib/models/assignment.dart [hunk 4]
```diff
@@ -68,10 +69,10 @@ abstract class Assignment implements Built<Assignment, AssignmentBuilder> {
 
   @nullable
   @BuiltValueField(wireName: 'submission')
-  BuiltList<Submission> get submissionList;
+  SubmissionWrapper get submissionList;
 
   Submission submission(String studentId) =>
-      submissionList?.firstWhere((submission) => submission.userId == studentId, orElse: () => null);
+      submissionList?.submissionList?.firstWhere((submission) => submission.userId == studentId, orElse: () => null);
```
[the remark this answers is not in the document]
Comment (Blackchai) [reply apps/flutter_parent/lib/models/assignment.dart:75]: Refer to the case above. It seems that if the call is made by the observer, we get a list, and if its for a student, we get a single submission. I'll do some further testing tho and possibly add in a fallback for each case.

File: apps/flutter_parent/lib/screens/calendar/calendar_widget/calendar_filter_screen/calendar_filter_list_screen.dart (hunk 1 above)
Comment (Blackchai) [reply apps/flutter_parent/lib/screens/calendar/calendar_widget/calendar_filter_screen/calendar_filter_list_screen.dart:112]: Oh, I hadn't seen that collection helper function. I'll try it out, thanks!

**Pull request merged**